src/vcf/validate_optional_policy.cpp

```diff
@@ -146,6 +152,44 @@ namespace ebi
         }
     }
 
+    void ValidateOptionalPolicy::check_body_entry_alt_gvcf_gt_value(ParsingState & state, Record const & record) const
+    {
+        if (std::find(record.alternate_alleles.begin(), record.alternate_alleles.end(), GVCF)
+            != record.alternate_alleles.end() && record.format[0] == vcf::GT) {
+            for (auto & sample : record.samples) {
+                if (sample_has_reference_in_all_alleles(sample)) {
+                    return;
+                }
+            }
+            throw new AlternateAllelesBodyError{state.n_lines,
+                    "At least one sample should contain genotype with all reference alleles, when ALT is " + GVCF
+                    + " as it is supposed to be a reference region"};
+        }
+    }
+
+    bool ValidateOptionalPolicy::sample_has_reference_in_all_alleles(std::string const & sample) const
+    {
+        std::string gt_subfield = sample.substr(0, sample.find(':'));
+        std::vector<std::string> alleles;
+        util::string_split(gt_subfield, "/|", alleles);
+        for (auto & allele : alleles) {
+            if (allele != "0") {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    void ValidateOptionalPolicy::check_body_entry_info_gvcf_end(ParsingState & state, Record const & record) const
+    {
+        if (std::find(record.alternate_alleles.begin(), record.alternate_alleles.end(), GVCF)
+            != record.alternate_alleles.end() && record.info.find(END) == record.info.end()) {
+            throw new InfoBodyError{state.n_lines,
+                    "INFO END should be provided, as " + GVCF
+                    + " (symbolic alternate allele) is supposed to represent a region"};
```

Comment:
For homogeneity with the other GVCF related error:

```
throw new InfoBodyError{state.n_lines,
        "INFO END should be provided when ALT is " + GVCF + " as it is supposed to be a region"};
```

I have avoided calling it a "reference region" to highlight the fact that is a region, plain and simple.